Ot285 142 firebase logout

Diego-otr opened the pull request.

nahuelba committed:
logic admin document with firestore added

nahuelba committed:
created selector of getUser in autheffects
app component html render layout depends if user is admin

nahuelba committed:
Merge branch 'main' into OT285-141-protect-private-routes

nahuelba committed:
Login with email and password added (firebase)

nahuelba committed:
Merge branch 'main' into OT285-141-protect-private-routes

Diego-otr committed:
header is hides logout btn if not logged

Diego-otr committed:
logOut: firebase signOut, set state to null

Diego-otr committed:
header/public-header, logout btn functionality

Diego-otr committed:
Merge branch 'main' into OT285-142-firebase-logout

Diego-otr committed:
logOut redirect to "auth/login"

Diego-otr committed:
Merge branch 'OT285-142-firebase-logout' into OT285-142-firebase-logout

carolinalaborde commented on src/app/core/components/header/header.component.html:
```diff
@@ -4,7 +4,7 @@
     <span class="title" [routerLink]="['backoffice/home']">Logo ONG</span>
     <span class="spacer"></span>
-    <button mat-button class="border">
+    <button *ngIf="logged$" mat-button class="border" (click)="logOut()">
```
async pipe?

carolinalaborde commented on src/app/core/components/public-header/public-header.component.ts:
```diff
@@ -50,15 +50,16 @@ export class PublicHeaderComponent implements OnInit {
-    console.log('Hacer el log out')
+  logOut(){
+    this.authService.logOut()
   }
```
Acá se debería hacer un dispatch de la acción logout y el servicio llamarse en un effect

carolinalaborde commented on src/app/features/services/auth.service.ts:
```diff
@@ -40,6 +45,13 @@ export class AuthService {
+    signOut(this.auth).then(() => {
+      this.store.dispatch(AuthPageActions.logOut());
+      this.router.navigate(['auth/login']);
+    });
```
No se deben hacer dispatch dentro del servicio.
Falta implementar el effect.
Podes ver como quedo implementado el login

Diego-otr committed:
logOut in auth.service fix
logout added in auth.effects
dispatch action moved to public header
header: async

Diego-otr committed:
logOut redirects to 'auth/login'
header component now dispatchs action

Diego-otr committed:
auth.service logOut type

Diego-otr committed:
Merge branch 'main' into OT285-142-firebase-logout

luccio93 committed:
Fix logOut. Add redirect after login/logout

Diego-otr committed:
Merge branch 'main' of https://github.com/alkemyTech/OT285-client into OT285-142-firebase-logout

Diego-otr committed:
rmv header useless router.navigate

Diego-otr committed:
Merge branch 'main' into OT285-142-firebase-logout
